server/src/main/java/org/elasticsearch/readiness/ReadinessService.java

```diff
@@ -253,7 +253,11 @@ private void setReady(boolean ready) {
      * Add a listener for bound readiness service address.
      * @param listener
      */
-    public void addBoundAddressListener(BoundAddressListener listener) {
+    public synchronized void addBoundAddressListener(BoundAddressListener listener) {
```

Comment:
Slightly unrelated - maybe worth adding a comment that `setupSocket` is expected to be called within the lock? Or Just make `setupSocket` synchronized...

Reply:
Added a comment.